addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/UserAgentScanRule.java

```diff
@@ -196,11 +198,23 @@ private boolean isBodyDifferent(HttpMessage newMsg) {
     }
 
     private void createAlert(HttpMessage newMsg, String userAgent) {
-        newAlert()
+        buildAlert(newMsg, userAgent).raise();
+    }
+
+    private AlertBuilder buildAlert(HttpMessage msg, String userAgent) {
+        return newAlert()
                 .setConfidence(Alert.CONFIDENCE_MEDIUM)
                 .setParam(USER_AGENT_PARAM_NAME)
                 .setAttack(userAgent)
-                .setMessage(newMsg)
-                .raise();
+                .setMessage(msg);
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<>();
+        Alert example = buildAlert(null, "ExampleBot 1.1").build();
+        example.setTags(CommonAlertTag.toMap(CommonAlertTag.CUSTOM_PAYLOADS));
```

Comment:
IMO still better to use the other method, less error prone if we add tags to the raised alerts.

Reply:
OH so in this case still use example.getTags() even though it will return empty, and merge it with CUSTOM_PAYLOADS?

Reply:
Yes.